pilot/pkg/model/network.go

```diff
@@ -482,11 +491,19 @@ func (n *networkGatewayNameCache) resolveAndCache(name string) []string {
 		ttl = MinGatewayTTL
 	}
 	expiry := time.Now().Add(ttl)
+	refreshPeriod := ttl
+	if len(addrs) == 0 {
+		// resolution failed, retry after 5s (by default)
+		// ttl is set to ~136 years
```

Comment:
I am not sure I understand where 136y is from or why

Reply:
It's the _default_ TTL
https://github.com/istio/istio/blob/1.17.1/pilot/pkg/model/network.go#L528
So if resolution failed, the TTL here will be 4294967295/86400/365 ~= 136